### ISSUE TITLE
**Image data augmentations**

### ISSUE THREAD

**Issue by arampacha** [arampacha/CLIP-rsicd#6]
- [ ] - select augmentations set
- [ ] - make aug pipeline

**Comment by sujitpal** [comment]
There is already a spot in the [training script](https://github.com/arampacha/CLIP-rsicd/blob/master/flax_clip_training_wip.py#L197) where the images are being sent through a variety of transformations.

- ToTensor
- Resize
- CenterCrop
- Normalize

but none of them are for augmentation, we can add a few here for that. Some candidates from [the torchvision docs](https://pytorch.org/vision/stable/transforms.html) are:

- RandomHorizontalFlip
- RandomVerticalFlip
- RandomCrop
- RandomAffine
- RandomResize
- etc.

I will work on parameters for these tomorrow and add them into the training script.

**Comment by sujitpal** [comment]
I have a set of augmenting transforms in the last cell of the [nbs/image-augmentation-explore.ipynb notebook](https://github.com/arampacha/CLIP-rsicd/blob/master/nbs/image-augmentation-explore.ipynb), which can be pushed into the Transforms code block in [flex_clip_training_wip.py](https://github.com/arampacha/CLIP-rsicd/blob/master/flax_clip_training_wip.py#L211).

Current block:
```
        self.transforms = torch.nn.Sequential(
            Resize(
                [
                    n_px,
                ],
                interpolation=InterpolationMode.BICUBIC,
            ),
            CenterCrop(n_px),
            ConvertImageDtype(torch.float),
            Normalize((0.48145466, 0.4578275, 0.40821073), (0.26862954, 0.26130258, 0.27577711)),
        )
```

propose replacing with:

```
        self.transforms = torch.nn.Sequential(
             # the first entry converts incoming Numpy array into PIL, may not be needed
             transforms.ToPILImage(), 
             transforms.RandomCrop(n_px),
             transforms.ColorJitter(),
             transforms.RandomHorizontalFlip(),
             transforms.RandomVerticalFlip(),
             transforms.RandomResizedCrop(n_px, scale=(0.8, 1.2), ratio=(1, 1)),
             transforms.RandomAdjustSharpness(1.5)
            // previous stuff
            Resize([n_px,], interpolation=InterpolationMode.BICUBIC),
            CenterCrop(n_px),
            ConvertImageDtype(torch.float),
            Normalize((0.48145466, 0.4578275, 0.40821073), (0.26862954, 0.26130258, 0.27577711)),
        )
```

Since the training file is probably being edited and don't want to cause merge conflicts, putting these changes in here so it can be picked up after the file is updated.
